app/models/concerns/course/search_concern.rb

```diff
@@ -0,0 +1,18 @@
+module Course::SearchConcern
+  extend ActiveSupport::Concern
+
+  module ClassMethods
+    # Search and filter courses by their titles, descriptions or user names.
+    # @param [String] keywords The keywords for filtering courses.
+    # @return [Array<Course>] The courses which match the keyword. All courses will be returned if
+    #   keyword is blank.
+    def search(keyword)
+      return all if keyword.blank?
+
+      condition = "%#{keyword}%"
+      joins { users.outer }.
+        where { (title =~ condition) | (description =~ condition) | (users.name =~ condition) }.
+        select { 'DISTINCT courses.*' }
```

Comment:
@lowjoel For your reference:
`$ Course.search('lol').order(:title)`

Use `select { 'DISTINCT courses.*' }`:
```
SELECT
        DISTINCT courses. *
FROM
         "courses" LEFT OUTER JOIN "course_users"
                ON "course_users" . "course_id" = "courses" . "id" LEFT OUTER JOIN "users"
        ON "users" . "id" = "course_users" . "user_id"
WHERE
        "courses" . "instance_id" = $1
        AND (
                (
                         "courses" . "title" ILIKE '%lol%'
                        OR "courses" . "description" ILIKE '%lol%'
                )
                OR "users" . "name" ILIKE '%lol%'
        )
ORDER BY
        "courses" . "title" ASC  [["instance_id", 1]]
```

Use `select { distinct('courses.*') }`:
```
PG::InvalidColumnReference: ERROR:  for SELECT DISTINCT, ORDER BY expressions must appear in select list:

SELECT 
        distinct('courses.*') 
FROM 
        "courses" LEFT OUTER JOIN "course_users" 
                ON "course_users"."course_id" = "courses"."id" LEFT OUTER JOIN "users" 
                ON "users"."id" = "course_users"."user_id" 
WHERE 
        "courses"."instance_id" = $1 
        AND (
                (
                "courses"."title" ILIKE '%lol%' 
                OR "courses"."description" ILIKE '%lol%'
                )
                OR "users"."name" ILIKE '%lol%'
        )  
ORDER BY 
        "courses"."title" ASC
```